src/overlays/actors/ovl_Door_Ana/z_door_ana.h

```diff
@@ -15,18 +15,17 @@ typedef struct DoorAna {
 
 extern const ActorInit Door_Ana_InitVars;
 
-#define DOORANA_TYPE_REGULAR      0x000
-#define DOORANA_TYPE_UNK          0x100
-#define DOORANA_TYPE_HIDDEN       0x200
-#define DOORANA_TYPE_ADJACENT     0x300
-#define DOORANA_TYPE_BITRANGE     0x300
-
-#define GET_DOORANA_TYPE(this)(this->actor.params & 0x300)
-
-#define DOORANA_TYPE_ROTATION_ENTRANCE(directEntrance)(directEntrance < 0)
-#define GET_DOORANA_DIRECT_ENTRANCE(this) ((((this)->actor.params >> 0xC) & 0x7) - 1) 
-#define GET_DOORANA_ADJACENT_ENTRANCE(this) ((this)->actor.params & 0x1F) 
-#define GET_DOORANA_ROTATION_ENTRANCE(this) (this->actor.home.rot.z + 1) 
-#define GET_DOORANA_ITEMFLAGS(this) ((this)->actor.params & 0xFF) 
+#define DOORANA_TYPE_VISIBLE            0x000
+#define DOORANA_TYPE_HIDDEN_STORMS      0x100 // Unused, leftover from OoT 
+#define DOORANA_TYPE_HIDDEN_BOMB        0x200
+#define DOORANA_TYPE_VISIBLE_SCENE_EXIT 0x300 // Destination dependent on scene exit list
+#define DOORANA_TYPE_BITRANGE           0x300
```

Comment:
```suggestion
#define DOORANA_TYPE_HIDDEN           DOORANA_TYPE_HIDDEN_STORMS | DOORANA_TYPE_HIDDEN_BOMB
```
Based on its usage, I think this makes more sense.

Reply:
strongly disagree. `DOORANA_TYPE` stores an enumeration, not a set of bitflags, and 0x0300 maps to `DOORANA_TYPE_VISIBLE_SCENE_EXIT` which shouldn't be called `DOORANA_TYPE_HIDDEN`.

The true purpose of the code that uses `DOORANA_TYPE_BITRANGE` is to force the type to become `DOORANA_TYPE_VISIBLE` in a fast but unconventional way.

Reply:
I agree. The 0x300 should map to `DOORANA_TYPE_VISIBLE_SCENE_EXIT` and should stay. My suggestion was meant to just apply to the extra 0x300  `DOORANA_TYPE_BITRANGE`. 

I feel that renaming `DOORANA_TYPE_BITRANGE` to `DOORANA_TYPE_HIDDEN` and using the or of the hidden types shows that is changing to `DOORANA_TYPE_VISIBLE` in clear way. Just using the word bitrange gives no information at all. Though looking at it now, oring the hidden types, probably doesn't make sense since as you said the types really should be an enum not bit flags. 

I still think `DOORANA_TYPE_BITRANGE` should be renamed to `DOORANA_TYPE_HIDDEN`, possibly with a small comment that 0x300 encapsulates all the hidden types.

Reply:
I opted for a different implementation that should clear things up more.

`DOORANA_TYPE_BITRANGE` is meant to be used as the bitmask for the "Type" property within the grotto's params, so I've renamed it as such and added it to DOORANA_GET_TYPE(), as well as implemented a `DOORANA_SET_TYPE` macro.